patchwork-biomes/src/main/java/net/minecraftforge/common/BiomeManager.java

```diff
@@ -64,14 +69,61 @@ private OverworldClimate getClimate() {
 	public static class BiomeEntry extends WeightedPicker.Entry {
 		public final Biome biome;
 
+		public int field_76292_a; // evil hack
+
 		public BiomeEntry(Biome biome, int weight) {
 			super(weight);
 
+			this.field_76292_a = weight;
 			this.biome = biome;
 		}
 
 		private int getWeight() {
 			return this.weight;
 		}
 	}
+
+	// Biomes O' Plenty pokes Forge internals. Fun
+	private static ArrayList<BiomeEntry>[] biomes = setupBiomes();
```

Comment:
Best practice would be to type this as List, rather than ArrayList, we shouldn't care about the class of the object behind it.

Reply:
Wait does Forge type this as ArrayList? If so :concern:

Reply:
IIRC it was a custom implementation of ArrayList to track if the ArrayList had been modified

Reply:
BOP calls setupBiomes directly through reflection. It casts the result to a List, but since the actual Forge type extends ArrayList, I thought this would be better.

Reply:
Why not go all the way and implement the class from forge in case someone casts to it?

Reply:
True, that just didn't seem necessary. I'll do that later.